lib/clean_test.py

```diff
@@ -36,3 +37,58 @@ def test_remove_future_dates(self):
                 columns=columns,
             ),
         )
+
+
+class FoatIntStr(unittest.TestCase):
+    def test_float_to_int_str(self):
+        data = {"dates": ["1994", "2005", "2011", "1955"]}
+        df = pd.DataFrame(data)
+
+        df.loc[:, "dates"] = df.dates.astype(float)
+        df = df.pipe(float_to_int_str, ["dates"])
+
+        dfa = pd.DataFrame(data)
+        dfa.loc[:, "dates"] = dfa.dates.astype(str)
+
+        df = df[["dates"]]
+
+        dfa = dfa[["dates"]]
+
+        assert_frame_equal(df, dfa)
```

Comment:
- Rather than specifying `data["dates"]` as a list of strings and then converting them into floats, you can just specify them as a list of floats in the first place.
- This test could also test what happens when `float_to_int_str` processes more than one column.
- The primary reason why this function is even needed is because Pandas convert int columns with any `NA` value into float columns. You should insert some `np.nan` values and assert that they are turned into empty string.
- Since both `df` and `dfa` only have one column, `df = df[["dates"]]` and `dfa = dfa[["dates"]]` are no op.
- It is better if you specify the second argument to `assert_frame_equal` inline rather than elsewhere. That way people can see at a glance what is the expected output.

Here's how I would write this method just to solidify my previous points:
```python

class FloatToIntStrTestCase(unittest.TestCase):
    def test_float_to_int_str(self):
        """
        This is short but it tests multiple things:
        - float_to_int_str works on multiple columns
        - float_to_int_str only works on specified columns
        - float_to_int_str transforms float columns to integer strs
        - float_to_int_str leave int64 columns alone
        - float_to_int_str transforms "mixed" column (dtype object)
        """
        columns = ["id", "name", "age", "mixed"]
        assert_frame_equal(
            float_to_int_str(
                pd.DataFrame(
                    [
                        [3, "john", 27.0, 1973.0],
                        [4, "anne", 24.0, np.nan],
                        [5, "bill", np.nan, "abc"],
                    ],
                    columns=columns,
                ),
                ["id", "age", "mixed"],
            ),
            pd.DataFrame(
                [
                    [3, "john", "27", "1973"],
                    [4, "anne", "24", ""],
                    [5, "bill", "", "abc"],
                ],
                columns=columns,
            ),
        )
```